modules/images/webp-uploads/load.php

```diff
@@ -684,6 +684,135 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $metadata              Metadata of the attachment.
+ * @param array  $valid_mime_transforms List of valid mime transforms for current image mime type.
+ * @param string $file                  Path to original file.
+ * @param array  $main_images           Path of all main image files of all mime types.
+ * @param array  $subsized_images       Path of all subsized image file of all mime types.
+ *
+ * @return array                        Metadata with sources added.
+ */
+function webp_uploads_update_sources( $metadata, $valid_mime_transforms, $file, $main_images, $subsized_images ) {
+	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
+
+	foreach ( $valid_mime_transforms as $targeted_mime ) {
+		// Add sources to original image metadata.
+		$image_file = $main_images[ $targeted_mime ]['path'];
+
+		if ( ! file_exists( $image_file ) ) {
+			continue;
+		}
+
+		$metadata['sources'][ $targeted_mime ] = array(
+			'file'     => wp_basename( $image_file ),
+			'filesize' => filesize( $image_file ),
+		);
+
+		foreach ( $metadata['sizes'] as $size => $size_details ) {
+			// Add sources to resized image metadata.
+			$image_file = $original_directory . '/' . $subsized_images[ $targeted_mime ][ $size ]['file'];
+
+			if ( ! file_exists( $image_file ) ) {
+				continue;
+			}
+
+			$metadata['sizes'][ $size ]['sources'][ $targeted_mime ] = array(
+				'file'     => wp_basename( $image_file ),
+				'filesize' => filesize( $image_file ),
+			);
+		}
+	}
+
+	return $metadata;
+}
+
+/**
+ * Creates additional image formats when original image is edited.
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( null !== $override ) {
+		return $override;
+	}
+
+	if ( empty( webp_uploads_get_upload_image_mime_transforms()[ $mime_type ] ) ) {
+		return null;
+	}
+
+	add_filter(
+		'wp_update_attachment_metadata',
+		function ( $metadata, $post_meta_id ) use ( $post_id, $file, $mime_type, $image ) {
+			if ( $post_meta_id !== $post_id ) {
+				return $metadata;
+			}
+
+			$valid_mime_transforms = webp_uploads_get_upload_image_mime_transforms()[ $mime_type ];
+			$original_directory    = pathinfo( $file, PATHINFO_DIRNAME );
+			$filename              = pathinfo( $file, PATHINFO_FILENAME );
+			$allowed_mimes         = array_flip( wp_get_mime_types() );
+			$old_metadata          = wp_get_attachment_metadata( $post_id );
+			$resize_sizes          = array();
+			$main_images           = array();
+			$subsized_images       = array();
```

Comment:
Nit-pick, but having all these variables defined right here even though they're not used right away makes the code a bit hard to read. These are all local variables, so we don't need to put them all together, since they aren't used anywhere else anyway.

`$allowed_mimes`, `$main_images`, and `$subsized_images` are only used in the second `foreach` loop below, so I think it would be better for code readability and clarity on what these variables are for if we moved those three from here directly above the second `foreach`.